src/client/render/secondstage.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+Copyright (C) 2017 numzero, Lobachevskiy Vitaliy <numzer0@yandex.ru>
+Copyright (C) 2020 appgurueu, Lars Mueller <appgurulars@gmx.de>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "secondstage.h"
+#include "client/client.h"
+#include "client/shader.h"
+#include "client/tile.h"
+
+RenderingCoreSecondStage::RenderingCoreSecondStage(
+		IrrlichtDevice *_device, Client *_client, Hud *_hud) :
+		RenderingCoreStereo(_device, _client, _hud)
+{
+	initMaterial();
+}
+
+void RenderingCoreSecondStage::initMaterial()
+{
+	IWritableShaderSource *s = client->getShaderSource();
+	s->global_additional_headers = "#define SECONDSTAGE 1\n";
+	s->rebuildShaders(); // TODO remove this if possible, use shader const setters?
```

Comment:
No const setter can ever set a macro. And only a macro may be used to change variable definitions and likewise.

Reply:
Thanks for the clarification. I'd still prefer to not rebuild the shaders (just building them after the global additional headers are already set).